actors/miner/tests/change_beneficiary_test.rs

```diff
@@ -0,0 +1,411 @@
+    let mut info = h.get_info(&rt);
+    let pending_beneficiary_term = info.pending_beneficiary_term.unwrap();
+    assert_eq!(beneficiary_change.expiration, pending_beneficiary_term.new_expiration);
+    assert_eq!(beneficiary_change.quota, pending_beneficiary_term.new_quota);
+    assert_eq!(first_beneficiary_id, pending_beneficiary_term.new_beneficiary);
+
+    //revoking unapprovel proposal
+    h.change_beneficiary(
+        &mut rt,
+        h.owner,
+        BeneficiaryChange::new(h.owner, TokenAmount::zero(), ChainEpoch::from(0)),
+        Some(h.owner),
+    )
+    .unwrap();
+
+    info = h.get_info(&rt);
+    assert_eq!(None, info.pending_beneficiary_term);
+    assert_eq!(h.owner, info.beneficiary);
+
+    h.check_state(&rt);
+}
+
+#[test]
+fn successfully_immediately_change_back_to_owner_address_while_used_up_quota() {
+    let (mut h, mut rt) = setup();
+    let first_beneficiary_id = Address::new_id(999);
+
+    let quota = TokenAmount::from(100);
+    let expiration = ChainEpoch::from(200);
+    h.propose_approve_initial_beneficiary(
+        &mut rt,
+        first_beneficiary_id,
+        BeneficiaryTerm::new(quota.clone(), TokenAmount::zero(), expiration),
+    )
+    .unwrap();
+
+    h.withdraw_funds(&mut rt, h.beneficiary, &quota, &quota, &TokenAmount::zero()).unwrap();
+    h.change_beneficiary(
+        &mut rt,
+        h.owner,
+        BeneficiaryChange::new(h.owner, TokenAmount::zero(), ChainEpoch::from(0)),
+        Some(h.owner),
+    )
+    .unwrap();
+
+    let info = h.get_info(&rt);
+    assert_eq!(None, info.pending_beneficiary_term);
+    assert_eq!(h.owner, info.beneficiary);
+    assert_eq!(TokenAmount::zero(), info.beneficiary_term.quota);
+    assert_eq!(ChainEpoch::from(0), info.beneficiary_term.expiration);
+    assert_eq!(TokenAmount::zero(), info.beneficiary_term.used_quota);
+    h.check_state(&rt);
+}
+
+#[test]
+fn successfully_immediately_change_back_to_owner_while_expired() {
+    let (mut h, mut rt) = setup();
+    let first_beneficiary_id = Address::new_id(999);
+
+    let quota = TokenAmount::from(100);
+    let expiration = ChainEpoch::from(200);
+    h.propose_approve_initial_beneficiary(
+        &mut rt,
+        first_beneficiary_id,
+        BeneficiaryTerm::new(quota, TokenAmount::zero(), expiration),
+    )
+    .unwrap();
+
+    rt.set_epoch(201);
+    h.change_beneficiary(
+        &mut rt,
+        h.owner,
+        BeneficiaryChange::new(h.owner, TokenAmount::zero(), ChainEpoch::from(0)),
+        Some(h.owner),
+    )
+    .unwrap();
+
+    let info = h.get_info(&rt);
+    assert_eq!(None, info.pending_beneficiary_term);
+    assert_eq!(h.owner, info.beneficiary);
+    assert_eq!(TokenAmount::zero(), info.beneficiary_term.quota);
+    assert_eq!(ChainEpoch::from(0), info.beneficiary_term.expiration);
+    assert_eq!(TokenAmount::zero(), info.beneficiary_term.used_quota);
+    h.check_state(&rt);
+}
+
+#[test]
+fn successfully_increase_quota() {
```

Comment:
What happens if they try to decrease the quota? What happens the beneficiary uses more than the smaller amount while the proposal to decrease is in-flight?

My best guess from the code is that the changing the quota to decrease it will work, and then we'll end up with a quota where used > quota. Could you demonstrate this with a test? If the decrease is allowed we should remove the "monotonic" note from the comment on BeneficiaryTerm.

We could disallow decreasing quota, but that seems unnecessarily restrictive.

Reply:
yes, you guess is right,  decrease quota was allow, confirm the final balance check is OK. I will add a test for this case.

Reply:
add test successfully_change_quota_and_test_withdraw